ui/app/serializers/task-group-deployment-summary.js

```diff
@@ -0,0 +1,9 @@
+import ApplicationSerializer from './application';
+
+export default ApplicationSerializer.extend({
+  normalize(typeHash, hash) {
+    hash.PlacedCanaryAllocations = hash.PlacedCanaries || [];
```

Comment:
Do you need this if you are setting the defaultValue above? (or vice versa)

Reply:
`defaultValue` isn't invoked when the value is `null` (as opposed to `undefined`), so this is needed. I suppose the `defaultValue` in the model isn't strictly necessary, but I like that it kinda doubles as a type annotation.